pkg/rthooks/args.go

```diff
@@ -0,0 +1,146 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Tetragon
+
+package rthooks
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+	"time"
+
+	v1 "github.com/cilium/tetragon/api/v1/tetragon"
+	"github.com/cilium/tetragon/pkg/cgroups"
+	"github.com/cilium/tetragon/pkg/watcher"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	uidStringLen = len("00000000-0000-0000-0000-000000000000")
+)
+
+type CreateContainerArg struct {
+	Req     *v1.CreateContainer
+	Watcher watcher.K8sResourceWatcher
+
+	// cached values
+	cgroupID *uint64
+	pod      *corev1.Pod
+}
+
+func (arg *CreateContainerArg) CgroupID() (uint64, error) {
+	if p := arg.cgroupID; p != nil {
+		return *p, nil
+	}
+
+	// retrieve the cgroup id from the host cgroup path.
+	//
+	// NB(kkourt): A better solution might be to hook into cgroup creation routines and create a
+	// mapping between directory and cgroup id that we maintain in user-space. Then, we can find
+	// the id using this mapping.
+	cgPath := arg.Req.CgroupsPath
+	cgRoot, err := cgroups.HostCgroupRoot()
```

Comment:
Not related to this one, but we could cache the cgroup root, we already do that in other parts: https://github.com/cilium/tetragon/blob/c2f3dbd92890903780a2269e96d5a47ca5713b50/pkg/cgroups/cgroups.go#L165  and it won't change, so this part of code can also just cache it.

Reply:
That makes sense to me. There are a couple of more things that I would like to improve the cgroups package. I will leave this as a followup.